cores/arduino/stm32/twi.c

```diff
@@ -539,7 +539,7 @@ void HAL_I2C_ListenCpltCallback(I2C_HandleTypeDef *hi2c)
 
   if((obj->i2c_onSlaveReceive != NULL) &&
     (obj->slaveMode == SLAVE_MODE_RECEIVE)) {
-    nbData = I2C_TXRX_BUFFER_SIZE - obj->handle.XferCount;
+    nbData = I2C_TXRX_BUFFER_SIZE - obj->handle.XferSize;
```

Comment:
Aren't you using HAL_I2C_SlaveRxCpltCallback and HAL_I2C_SlaveTxCpltCallback ?

Reply:
no, only HAL_I2C_ListenCpltCallback

Reply:
Ok. then it's not easy for me to understand the code. If this works - fine. But I think  ListenCpltCallback does not mean that data has been transmitted already, so I don't understand the lines above and why XferCount is replaced by XferSize

Reply:
Right, I think, that we missed that when @fprwi6labs do the #135.
While I2C is sequential it seems to be OK but this will need to be review to handle that properly.